test/predicates.jl

```diff
@@ -189,6 +189,21 @@
     @test intersects(t, r1) == true
     @test intersects(r2, t) == false
 
+    s = Sphere(P2(0,3),1)
+    r =  Ray((0,0),(1,0))
+    @test intersects(r,s) == false
+    @test intersects(s,r) == false
+    r =  Ray((0,0),(0,1))
+    @test intersects(r,s) == true
+    @test intersects(s,r) == true
+    s = Sphere(Point3(0,3,0),1)
+    r =  Ray((0,0,0),(1,0,0))
+    @test intersects(r,s) == false
+    @test intersects(s,r) == false
+    r =  Ray((0,0,0),(0,1,0))
+    @test intersects(r,s) == true
+    @test intersects(s,r) == true
```

Comment:
We need more less trivial tests. For example, rays that are not aligned with the coordinate reference system, rays and balls in all 4 quadrants (2D) or 8 octants (3D).

Reply:
I could write a way to generate a collection of tests, i.e.
```julia 
using Rotations: Angle2d

r = Ray((0,0),(1,0))
s_t = Sphere(P2(3, 0), 1)
s_f = Sphere(P2(3, 3), 1)

translations = (Translate(p) for p in [(0,0), (10,0), (0,10), (-10,0), (0,-10))
rotations = (Rotate(Angle2d(theta) for theta in [0 : pi/4 : 7pi/4])
for t in translations, rot in rotations
    r_ =  t(rot(r))
    s_t_ =t(rot(s_t)) 
    s_f_ =t(rot(s_f)) 
    @test intersects(r_, s_t_)
    @test !intersects(r_, s_f_)
end
```
Perhaps that's a bit much, but it would cover all the bases. I can write something similar to for the 3D case, it would just have more translations and rotations to cover all the octants and rotation matrices.

Reply:
Actually, now that I think about it, that could be made into a function and be used for literally any `intersects` method, as long as you provide a positive example and negative example of the intersections you want to test in various locations and orientations.

Reply:
A function or a loop with tests seems like a good idea. Just make sure you avoid underscores in the variable names according to the code style of the project.